boards/FSM/Src/App/App_Brake.c

```diff
@@ -0,0 +1,70 @@
+#include <stdlib.h>
+#include <assert.h>
+#include "App_SharedBinaryStatus.h"
+#include "App_InRangeCheck.h"
+
+struct Brake
+{
+    struct InRangeCheck *brake_pressure_in_range_check;
+    struct BinaryStatus *is_brake_open_or_short_circuit_binary_status,
+        *is_brake_actuated_binary_status;
+
+    float min_brake_pressure_psi;
+    float max_brake_pressure_psi;
+
+    float (*get_pressure_psi)(void);
+    bool (*is_open_or_short_circuit)(void);
+    bool (*is_brake_actuated)(void);
+};
+
+struct Brake *App_Brake_Create(
+    float min_brake_pressure_psi,
+    float max_brake_pressure_psi,
+    float (*get_pressure_psi)(void),
+    bool (*is_open_or_short_circuit)(void),
+    bool (*is_brake_actuated)(void))
+{
+    struct Brake *brake = malloc(sizeof(struct Brake));
+    assert(brake != NULL);
+
+    brake->brake_pressure_in_range_check = App_InRangeCheck_Create(
+        get_pressure_psi, min_brake_pressure_psi, max_brake_pressure_psi);
+    brake->is_brake_open_or_short_circuit_binary_status =
+        App_SharedBinaryStatus_Create(is_open_or_short_circuit);
+    brake->is_brake_actuated_binary_status =
+        App_SharedBinaryStatus_Create(is_brake_actuated);
+
+    brake->min_brake_pressure_psi = min_brake_pressure_psi;
+    brake->max_brake_pressure_psi = max_brake_pressure_psi;
+
+    brake->get_pressure_psi         = get_pressure_psi;
+    brake->is_open_or_short_circuit = is_open_or_short_circuit;
+    brake->is_brake_actuated        = is_brake_actuated;
+
+    return brake;
+}
+
+void App_Brake_Destroy(struct Brake *brake)
+{
+    free(brake);
+}
+
+void App_Brake_GetBrakePressureInRangeCheck(
```

Comment:
This function doesn't seem to make sense, I can't tell if you're  trying to return the struct InRangeCheck (based on the function name) or enum InRangeCheck_Status. In either case, currently this returns void which just doesn't make sense to me.